src/mscorlib/src/System/Environment.cs

```diff
@@ -703,6 +693,9 @@ private static void SetEnvironmentVariableCore(string variable, string value)
                         // The error message from Win32 is "The filename or extension is too long",
                         // which is not accurate.
                         throw new ArgumentException(SR.Format(SR.Argument_LongEnvVarValue));
+                    case Win32Native.ERROR_NOT_ENOUGH_MEMORY:
+                    case Win32Native.ERROR_NO_SYSTEM_RESOURCES:
+                        throw new OutOfMemoryException(Interop.Kernel32.GetMessage(errorCode));
```

Comment:
I wonder if InsufficientMemoryException might be better, it is more explicit that the state of the process should not be corrupt at this point.

Reply:
It looks like that's only used by MemoryFailPoint. Also, this should not be hit in any normal case.